File: leetcode/0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count=0,cand=0;
        for(auto c: nums){
            if(count==0){
                cand=c;
                count++;
            }
            else if(cand==c){
                count++;
            }else{
                count--;
            }
        }
        return cand;
        
    }
};

File: leetcode/0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> mp=new HashMap<>();
        for(int x:nums){
            mp.put(x,mp.getOrDefault(x,0)+1);
            if(mp.get(x)>nums.length/2){
                return x;
            }
        }
        return -1;
    }
}